src/math/generic/fmod.rs

```diff
@@ -1,84 +1,63 @@
-/* SPDX-License-Identifier: MIT */
-/* origin: musl src/math/fmod.c. Ported to generic Rust algorithm in 2025, TG. */
-
 use super::super::{CastFrom, Float, Int, MinInt};
 
+/// Given the bits of a positive float, clamp the exponent field to [0,1]
+fn collapse_exponent<F: Float>(bits: F::Int) -> F::Int {
+    let sig = bits & F::SIG_MASK;
+    if sig == bits { sig } else { sig | F::IMPLICIT_BIT }
+}
+
+/// Computes (x << e) % y
```

Comment:
```suggestion
/// Computes `(x << e) % y` exactly without using bigint math
```

Or something to indicate why this isn't exactly `(x << e) % y`